libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/http/HttpDriver.java

```diff
@@ -258,7 +259,9 @@ public String doGet(final String url) throws IOException {
      */
     public String doPost(HttpPost httpPost) throws IOException {
         String url = httpPost.getURI().toString();
-        String query = EntityUtils.toString(httpPost.getEntity());
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        httpPost.getEntity().writeTo(byteArrayOutputStream);
+        String query = byteArrayOutputStream.toString();
```

Comment:
where are we using this object - I only see two references in log debug and error points. In which case why are we doing this here at all? We can do a private method to return the query and then have something like:

if (LOGGER.isDebugEnabled()) {
LOGGER.debug(...., getPostQuery(httpPost);
}

and then further down:

} catch (IOException ex) {
     LOGGER.error(......, getPostQuery(httpPost), ...);
}

Reply:
Thanks you're right I should have thought about it. It will also enable me to cleanly surround it with try-catch so if it will fail in the future it won't stop the client from sending the message.